app/controllers/trains_controller.rb

```diff
@@ -58,4 +60,12 @@ def train_params
   def train
     @train
   end
+
+  def set_active
+    if @train["active"] == "true"
+      @train["active"] = true
+    else
+      @train["active"] = false
```

Comment:
hmm id argue this is a code smell. somethings wrong if we have to be writing code like this.
what are code smells, well id recommend watching this video: https://www.youtube.com/watch?v=PJjHfa5yxlU
sandi metz is a great presenter on a huge range of topics.

An alternative to this, after ive done some digging is we stop passing a raw hash to our forms and instead pass an object.

what we can do is where we are setting things like
```
@train = TrainApi.new.show(params[:id])
```

we could do something like
```
@train = OpenStruct.new(TrainApi.new.show(params[:id]))
```

OpenStruct allows us to take a hash and turn it into an object so we get back cool accessors like
`@train.active` rather then doing `@train["active"]`
if we do something like this we get a lot of helpers back to help the setting of values across our form.

now ofcourse with this example
```
@train = OpenStruct.new(TrainApi.new.show(params[:id]))
```
you would have to modify everywhere you set `@train` to set them up a struct.

Doing something like using OpenStruct imo makes the whole thing feel a little better, opposed to taking and changing params back to different values based on what was submitted